### PR TITLE
Migrate to controller-runtime

### LINKED ISSUE #545: Migrate machine-controller to use controller-runtime
To align with other controllers, reduce duplicate caches, simplify the code and make testing easier, we should migrate the machine-controller to use controller-runtime.

AC:

* The  machine-controller uses controller-runtime

/kind cleanup

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #545 

**Special notes for your reviewer**:

```release-note
none
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@mrIncompetent PTAL

File: pkg/clusterinfo/configmap.go [hunk 1]
```diff
@@ -37,18 +37,18 @@ const (
...
 	}
 }
 
 type KubeconfigProvider struct {
-	clientConfig    *rest.Config
-	configMapLister listerscorev1.ConfigMapLister
-	endpointLister  listerscorev1.EndpointsLister
+	clientConfig *rest.Config
+	// We use a kubeClient to not accidentally create a lister
+	// in the ctrlruntimeclient
```
Comment (mrIncompetent) [inline pkg/clusterinfo/configmap.go]: Why is that bad? -> Comment

**Comment by mrIncompetent** [comment]
Except for a [nit](https://github.com/kubermatic/machine-controller/pull/592#discussion_r305795239), looks good :)

**Comment by alvaroaleman** [comment]
@mrIncompetent Updated, PTAL

**Pull request merged**